pkg/tunneler/listener.go

```diff
@@ -0,0 +1,290 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tunneler
+
+// Based on https://github.com/golang/build/blob/master/tunneler/v2/tunneler.go
```

Comment:
worth to comment why we need a fork?

Reply:
is not really a fork, I took only the idea and the algorithm, but the code is  different, I just wanted to give credit

Reply:
ack